lib/OptimizationManopt/src/OptimizationManopt.jl

```diff
@@ -0,0 +1,306 @@
+                              retraction_method::AbstractRetractionMethod = default_retraction_method(M),
+                              vector_transport_method::AbstractVectorTransportMethod = default_vector_transport_method(M),
+                              stepsize = WolfePowellLinesearch(M;
+                                                               retraction_method = retraction_method,
+                                                               vector_transport_method = vector_transport_method,
+                                                               linesearch_stopsize = 1e-12))
+    QuasiNewtonOptimizer{typeof(eval), typeof(M), typeof(retraction_method),
+                         typeof(vector_transport_method), typeof(stepsize)}(M,
+                                                                            retraction_method,
+                                                                            vector_transport_method,
+                                                                            stepsize)
+end
+
+function call_manopt_optimizer(opt::QuasiNewtonOptimizer{Teval},
+                               loss,
+                               gradF,
+                               x0,
+                               stopping_criterion::Union{Nothing, Manopt.StoppingCriterion}) where {
+                                                                                                    Teval <:
+                                                                                                    AbstractEvaluationType
+                                                                                                    }
+    sckwarg = stopping_criterion_to_kwarg(stopping_criterion)
+    opts = quasi_Newton(opt.M,
+                                      loss,
+                                      gradF,
+                                      x0;
+                                      return_options = true,
+                                      evaluation = Teval(),
+                                      retraction_method = opt.retraction_method,
+                                      vector_transport_method = opt.vector_transport_method,
+                                      stepsize = opt.stepsize,
+                                      sckwarg...)
+    # we unwrap DebugOptions here
+    minimizer = Manopt.get_solver_result(opts)
+    return (; minimizer = minimizer, minimum = loss(opt.M, minimizer), options = opts),
+           :who_knows
+end
+
+## Optimization.jl stuff
+
+function build_loss(f::OptimizationFunction, prob, cur)
+    function (::AbstractManifold, θ)
+        x = f.f(θ, prob.p, cur...)
+        __x = first(x)
+        return prob.sense === Optimization.MaxSense ? -__x : __x
+    end
+end
+
+function build_gradF(f::OptimizationFunction{true}, prob, cur)
+    function (::AbstractManifold, G, θ)
+        X = f.grad(G, θ, cur...)
```

Comment:
@mateuszbaran this needs to be wrapped in ManifoldDiff.gradient before passing to the optimizer, right?

Reply:
I was a little confused when using Manopt.jl directly as well, so probably I am misunderstanding the interface?

Reply:
I would have to check exactly, what `prob` is, but it seems to be an `AbstractManoptProblem` ? 

That one stores the Riemannian gradient, which does not require the conversion (from Euclidean to Riemannian gradient) that `ManifoldDiff.gradient` provides.

In Manopt.jl, if you provide `f` and `grad_f` the gradient is the Riemannian one (by default), so no conversion necessary (that would also be quite a hustle to always think of that in between).

You _can_ specify explicitly, that your gradient is Euclidean (also the Hessian then). Then your gradient (let‘s give it a different name) `grad_eucl_f` is “wrapped” in the gradient conversion before stored in the problem.
Then this gradient conversion happens every gradient call internally, but the Manopt problem still stores the Riemannian gradient.

For all details see the tutorial at https://manoptjl.org/stable/tutorials/EmbeddingObjectives/ (where I just see I might have broken the Hessian in that example somewhen in between, will check).

Reply:
Yes, as Ronny wrote, this entirely depends on what kind of gradient `f.grad` computes. I'd say that conversion between Euclidean and Riemannian gradient isn't something that should happen in the Manopt wrapper but either in the definition of optimization function or in a more general part of the package if you consider it useful for other solvers.

Reply:
Thanks @kellertuer that does clear it up, so I wasn't misunderstanding it, that's good 😄 

@mateuszbaran I think for now I'll keep the conversion in the wrapper since I don't see any other work happening in implementing these solvers outside Manopt? I think that'll be easier to maintain as well

Reply:
> @mateuszbaran I think for now I'll keep the conversion in the wrapper since I don't see any other work happening in implementing these solvers outside Manopt? I think that'll be easier to maintain as well

That's totally fine. Technically Optim.jl has `Manifold` type, NLLSsolver.jl has some support for manifold variables, and so does OptimKit.jl but I haven't seen anything as comprehensive as Manopt.jl.